### ISSUE TITLE
import-control module doesn't complain about disallowed 'fully qualified name' usages

### ISSUE THREAD

**Issue by qoomon** [checkstyle/checkstyle#6486]
import-control module doesn't complain about disallowed 'fully qualified name' usages
any possibility to check for illegal package access from a class to another?

**Comment by rnveach** [comment]
@qoomon Please see http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug.3F .
I am not fully sure what you are asking to give you a good response. An example would help.

**Comment by qoomon** [comment]
When I have the following ImportControl config and Java file, no validation is detected for the fully qualified name usage of `new java.util.LinkedList<String>();` although I've explicitly disallowed usage of `java.util` package.
However it works when I use `import java.util.LinkedList;` a validation is detected as expected.

```xml
<!DOCTYPE import-control PUBLIC
        "-//Checkstyle//DTD ImportControl Configuration 1.4//EN"
        "https://checkstyle.org/dtds/import_control_1_4.dtd">

<import-control pkg="com.qoomon.java.sandbox" strategyOnMismatch="disallowed">
    <disallow pkg="java.util"/>
</import-control>
```

```java
public class App {
    public static void main(String[] args) {
        new java.util.LinkedList<String>();
    }
}
```

**Comment by romani** [comment]
@qoomon ,

please STRICTLY follow defined format of issue. Issues that are not following format will be closed in 1 month after creation.

**Comment by rnveach** [comment]
Here is format:

````
$ cat TestClass.java
public class App {
    public static void main(String[] args) {
        new java.util.LinkedList<String>();
    }
}

$ cat TestImport.xml
<!DOCTYPE import-control PUBLIC
        "-//Checkstyle//DTD ImportControl Configuration 1.4//EN"
        "https://checkstyle.org/dtds/import_control_1_4.dtd">

<import-control pkg="com.qoomon.java.sandbox" strategyOnMismatch="disallowed">
    <disallow pkg="java.util"/>
</import-control>

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="ImportControl">
            <property name="file" value="TestImport.xml" />
        </module>
    </module>
</module>

$ java -jar checkstyle-8.18-all.jar -c TestConfig.xml TestClass.java
Starting audit...
Audit done.
````

He wants a violation even though there is no import.
As check name suggests, it only validates on imports which is sad since the import control contains almost all the rules for a project when it is done correctly.

We would need some other check to forbid using fully qualified class but there isn't a clean way for Checkstyle to do this as we don't know if `java` is a package or a class/field/variable.

**Comment by qoomon** [comment]
@rnveach exactly, that is what I want. And thx for explaining why it's complicated to implement.

**Comment by romani** [comment]
I doubt we can fix it, as Check is about import control, type usage is not done through imports.

Fully qualified names for some types should be forbidden by some other Check, we do not have existing Check for this

**Comment by rnveach** [comment]
I believe you are talking about this check
http://checkstyle.sourceforge.net/config_coding.html#IllegalType

**Comment by qoomon** [comment]
This solves the problem only partially, cause it is not possible to disallow whole packages at once.

**Comment by rnveach** [comment]
Someone else requested a similar addition to this Check to allow regular expressions to ban an entire package. See https://github.com/checkstyle/checkstyle/issues/5900#issuecomment-435654653 .

It was never responded to if it was a viable option to do this. @romani What do you think?

**Comment by romani** [comment]
Ban of whole package is better to do by some other Check new or existing. IllegalType should care about certain types only.
